### PR TITLE
feat(NcAppNavigation): add `n` hotkey to toggle navigation

### PR DESCRIPTION
**Pull request by skjnldsv**
Subset of https://github.com/nextcloud/server/issues/30484

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
**n**ice

File: package.json [hunk 1]
```diff
@@ -99,6 +99,7 @@
     "splitpanes": "^2.4.1",
     "string-length": "^5.0.1",
     "striptags": "^3.2.0",
+    "tabbable": "^6.2.0",
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply package.json:102]: Yes, but still more clear to have it included :)

**Comment by skjnldsv** [comment]
> Could you also add `aria-keyshortcuts` to the toggle button for accessibility

Sure :)

> Should we also close navigation by n?

We could do escape instead, assuming if we're in focus within the navigation :thinking:

**Comment by skjnldsv** [comment]
Done @ShGKme 
Also added cypress tests

**Comment by skjnldsv** [comment]
> Only in the mobile view, when the navigation is a popover

I was already done then.

File: src/components/NcAppNavigationToggle/NcAppNavigationToggle.vue [hunk 2]
```diff
@@ -50,9 +53,15 @@ export default {
...
+	setup() {
+		return { disableKeyboardShortcuts }
+	},
+
 	computed: {
 		label() {
-			return this.open ? t('Close navigation') : t('Open navigation')
+			return this.open
+				? t('Close navigation')
+				: t('Open navigation {shortcut}', { shortcut: disableKeyboardShortcuts ? '' : '[n]' })
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/NcAppNavigationToggle/NcAppNavigationToggle.vue]: but it's not compatible with rtl languages then :)

**Comment by skjnldsv** [comment]
> It is done for popover in mobile view via ESC, but what about n as a navigation toggle in general case?

I don't get it. What's the issue?

**Comment by skjnldsv** [comment]
> My proposal is to add n hotkey to **toggle** navigation (open when closed, close when opened).

Initially it did add toggle :see_no_evil: 
I changed it because I thought you wanted esc instead ahha

**Comment by skjnldsv** [comment]
Done @ShGKme 

- if closed, pressing n will open it
- if opened AND focus within, pressing n will close it
- if opened and focus NOT within, pressing n will do nothing

**Review by hamza221 — APPROVED** [review]

**Comment by skjnldsv** [comment]
> So there is no way to close navigation with a hotkey without putting the focus on the navigation first?

I think that would lead to issues otherwise

**Pull request merged**